configdialog.cpp

```diff
@@ -294,6 +294,20 @@ BOOL PaletteCustomizeDialog::command(WPARAM wparam, LPARAM lparam) {
 			refresh_palette_list();
 			break;
 		}
+		case IDC_PALDLG_IMPORT_EXPORT_PALETTES: {
+			std::wstring *palettes = (std::wstring *) DialogBoxParam(
+				NULL,
+				MAKEINTRESOURCE(DLG_IMPORT_EXPORT_PALETTES),
+				m_dialog,
+				(DLGPROC) ScreenSaverImportExportPalettesDialog,
+				(LPARAM) (export_palettes().c_str())
```

Comment:
⚠️ Because the result of `export_palettes()` is not assigned to anything, it is a temporary object and is destroyed immediately. `c_str()` returns a pointer to the string's underlying storage, but by the time this expression finishes evaluating, the string will have been destroyed and the pointer will be invalid. 
The result of `export_palettes` should be stored in a variable here so it has somewhere to live and won't instantly be killed.

Reply:
yeah, fair enough
given this, it's been working surprisingly well.